components/tidb_query_expr/src/impl_string.rs

```diff
@@ -980,6 +981,98 @@ fn substring(input: BytesRef, pos: Int, len: Int, writer: BytesWriter) -> Result
     Ok(writer.write_ref(Some(&input[start..end])))
 }
 
+#[rpn_fn(writer)]
+#[inline]
+pub fn substring_2_args_utf8(
```

Comment:
Maybe we can implement a `substring_utf8`? As `substring_2_args_utf8` is just a special case for `substring_3_args_utf8`.

Reply:
But it seems that tidb also use `substring_2_ars_utf8` and `substring_3_args_utf8`.